[ST] Add documentation - quickstart, running tests, config (#2685)
streamshub/console: systemtests/CONFIGURATION.md

| `SKIP_STRIMZI_INSTALLATION` | `false` | Skip installing Strimzi if it's already present in the cluster. |
| `STRIMZI_OPERATOR_NAME` | `strimzi-cluster-operator` | Strimzi deployment name — update when using an external Strimzi. |
| `CONSOLE_DEPLOYMENT_NAME` | `streamshub-console` | Console instance deployment name. |
| `CONSOLE_INSTALL_TYPE` | `olm` | `olm` (Operator Lifecycle Manager) or `yaml` (direct manifests). |
| `CLIENT_TYPE` | `kubectl` | `kubectl` or `oc` (OpenShift). |
| `CONSOLE_CLUSTER_DOMAIN` | _none — must be set_ | Ingress domain for non-OpenShift clusters. Auto-detected on OpenShift; otherwise used verbatim, with no auto-detection. For minikube, compute it per-cluster: `$(minikube ip).nip.io`. `config.yaml` ships `192.168.49.2.nip.io` as an example only — it's one specific minikube instance's IP and will not match yours. |

## OLM Configuration

| Variable | Default | Description |
|----------|---------|-------------|
| `CONSOLE_OLM_CATALOG_SOURCE_NAME` | `console-source` | CatalogSource name for the Console Operator. |

> | `CONSOLE_OLM_CATALOG_SOURCE_NAMESPACE` | `openshift-marketplace` | Namespace of the CatalogSource. Only correct on real OpenShift. On minikube/kind, OLM's catalog namespace is `olm` (e.g. after `minikube addons enable olm`) — CI overrides this variable to `olm` for those runs. |

Reviewer: Depending on the order of merge, this should be changed to `olm` in #2750 if this PR merges first, or change to `olm` here.

Author: Yea i'm getting a bit ahead of myself, i wanted to add that here in docs, expecting that they would be merged after https://github.com/streamshub/console/pull/2750 and https://github.com/streamshub/console/pull/2763

Author: But i can remove this part and add it into the respective PRs, that would be better.

RUN_HEADLESS: false
TEST_CONSOLE_LOG_LEVEL: DEBUG
```

**2. Set environment variables** (temporary overrides)
```bash
export RUN_HEADLESS=false
export TEST_CONSOLE_LOG_LEVEL=DEBUG
mvn -pl systemtests verify
```

Environment variables take precedence over config.yaml values.

## Configuration Categories

### Common Settings

General test behavior and logging configuration:

| Variable | Default | Description |
|----------|---------|-------------|
| `CLEANUP_ENVIRONMENT` | `true` | Clean up Kubernetes resources after each test. Set to `false` to inspect cluster state after test failures. |
| `DELETE_CONSOLE_OPERATOR_BEFORE_INSTALL` | `true` | Re-deploy Console operator if it already exists in the cluster. |
| `TEST_FILE_LOG_LEVEL` | `DEBUG` | Log level for file output. Options: `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`. |
| `TEST_CONSOLE_LOG_LEVEL` | `INFO` | Log level for console output. Keeps console clean while files have details. |
| `TEST_LOG_DIR` | `target/logs` | Directory path for test logs, YAML dumps, and diagnostic output. |
| `RUN_HEADLESS` | `true` | Run Playwright tests without visible browser window. Set to `false` to watch UI tests execute. |

> | `SCREENSHOTS_DIR_PATH` | `screenshots` | Directory for test failure screenshots. Relative to systemtests directory. |

Reviewer: Should the screenshots be dropped into `target` or `target/logs` by default? Some examples in the `TROUBLESHOOTING.md` show them placed there and it might be good to make them be removed automatically from `mvn clean`.

Author: My personal opinion on this is that for debug, it is from time to time useful to compare previous run traces with the current one, logs dynamically change, but the pw trace/screenshots look mostly the same and can point to specific ui issues persisting. But i also agree that cleaning them would be nice, because over the time their folders fill up and take space on your disk.

| `SKIP_STRIMZI_INSTALLATION` | `false` | Skip Strimzi installation if already installed externally. |
| `STRIMZI_OPERATOR_NAME` | `strimzi-cluster-operator` | Deployment name of Strimzi Cluster Operator. Update if using external Strimzi. |
| `CONSOLE_DEPLOYMENT_NAME` | `streamshub-console` | Deployment name of Console instance. |
| `CONSOLE_INSTALL_TYPE` | `olm` | Installation method: `olm` (Operator Lifecycle Manager) or `yaml` (direct manifests). |
| `CLIENT_TYPE` | `kubectl` | Kubernetes CLI client: `kubectl` or `oc` (for OpenShift). |

> | `CONSOLE_CLUSTER_DOMAIN` | `192.168.49.2.nip.io` | Cluster domain for ingress routes. Required for non-OpenShift clusters. Update to match your cluster IP. |

Reviewer: ```suggestion
| `CONSOLE_CLUSTER_DOMAIN` | N/A | Cluster domain for ingress routes. Required for non-OpenShift clusters. Update to match your cluster IP. |
```

Should we remove the default here? It's always going to be variable and having a default could lead to someone thinking that would automatically work,

### OLM Configuration

Settings for Operator Lifecycle Manager installation method:

| Variable | Default | Description |
|----------|---------|-------------|
| `CONSOLE_OLM_CATALOG_SOURCE_NAME` | `console-source` | Name of the CatalogSource for Console Operator. |

> | `CONSOLE_OLM_CATALOG_SOURCE_NAMESPACE` | `openshift-marketplace` | Namespace where CatalogSource is created. |

Reviewer: ```suggestion
| `CONSOLE_OLM_CATALOG_SOURCE_NAMESPACE` | `olm` | Namespace where CatalogSource is created. |
```

I think the OLM install scripts use `olm`, which may be a better default here.

| `CONSOLE_OLM_PACKAGE_NAME` | `streamshub-console-operator` | Package name in the operator catalog. |
| `CONSOLE_OLM_CHANNEL_NAME` | `alpha` | Subscription channel for selecting operator version. |

**When to use OLM:** Testing on OpenShift or when validating operator bundle/catalog functionality.

### YAML Installation Configuration

Settings for direct YAML manifest installation:

| Variable | Default | Description |
|----------|---------|-------------|
| `STRIMZI_OPERATOR_VERSION` | `1.0.0` | Version of Strimzi Cluster Operator to install. |
| `CONSOLE_OPERATOR_BUNDLE_URL` | `https://github.com/streamshub/console/releases/download/0.12.4/streamshub-console-operator.yaml` | URL or path to console-operator.yaml. Accepts: URL, relative path from project root, or absolute path. |

**When to use YAML:** Testing on non-OpenShift clusters (minikube, kind, k3s) or when testing specific operator builds.

### Console Instance Images

Override default images installed by the operator:

| Variable | Default | Description |
|----------|---------|-------------|
| `CONSOLE_API_IMAGE` | _(empty)_ | Custom Console API image. Overrides operator's default API image. |

> | `CONSOLE_UI_IMAGE` | _(empty)_ | Custom Console UI image. Overrides operator's default UI image. |

Reviewer: ```suggestion
```
Obsolete

**Use case:** Testing local changes to Console API or UI before they're released.

**Example:**
```yaml
CONSOLE_API_IMAGE: quay.io/your-username/console-api:my-feature

> CONSOLE_UI_IMAGE: quay.io/your-username/console-ui:my-feature

Reviewer: ```suggestion
```
Obsolete

| `KEYCLOAK_VERSION` | `26.4.0` | Keycloak (identity provider) release version. |
| `PROMETHEUS_VERSION` | `0.89.0` | Prometheus (metrics) operator release version. |
| `ST_KAFKA_VERSION` | `4.2.0` | Kafka version to use in test clusters. |
| `TEST_CLIENTS_IMAGE` | `quay.io/strimzi-test-clients/test-clients:0.14.0-kafka-4.2.0` | Container image for producer/consumer test clients. |

**Note:** Update versions when testing compatibility with newer component releases.

## Configuration Examples

### Debug Mode Configuration

For investigating test failures:

```yaml
CLEANUP_ENVIRONMENT: false       # Keep resources after test
RUN_HEADLESS: false              # Watch browser interactions
TEST_CONSOLE_LOG_LEVEL: DEBUG    # Detailed console output
TEST_FILE_LOG_LEVEL: TRACE       # Maximum file logging detail
```

### CI/Automation Configuration

For automated test execution:

```yaml
CLEANUP_ENVIRONMENT: true        # Always clean up
RUN_HEADLESS: true               # No GUI needed
TEST_CONSOLE_LOG_LEVEL: INFO     # Clean console output
```

### Custom Image Testing

Testing locally-built Console:

```yaml
CONSOLE_API_IMAGE: localhost:5000/console-api:dev

> CONSOLE_UI_IMAGE: localhost:5000/console-ui:dev

Reviewer: ```suggestion
```
Obsolete
